lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -492,10 +492,13 @@ def setUp(self):
         # sure if there's a way to poke the test harness to do so.  So, we'll
         # just inject the override field storage in this brute force manner.
         OverrideFieldData.provider_classes = None
+        # pylint: disable=protected-access
         for block in iter_blocks(course):
-            block._field_data = OverrideFieldData.wrap(   # pylint: disable=protected-access
-                coach, block._field_data)   # pylint: disable=protected-access
-            block._field_data_cache = {'tabs': [], 'discussion_topics': []}  # pylint: disable=protected-access
+            block._field_data = OverrideFieldData.wrap(coach, block._field_data)
+            new_cache = {'tabs': [], 'discussion_topics': []}
+            if 'grading_policy' in block._field_data_cache:
+                new_cache['grading_policy'] = block._field_data_cache['grading_policy']
+            block._field_data_cache = new_cache
```

Comment:
@cpennington because before this block it was maked as dirty (there're likely a read somewhere  before, but I haven't managed to find it) *and* `_field_data_cache` contained it. As a result, `grading_policy` was present in `_dirty_fields`, but missing from `_field_data_cache` - it crashed the test in `bind_for_student` - `course` descriptor is saved there, save searches for modified fields, it gets all the way down to `_is_dirty` and crashed because of missing key at https://github.com/edx/XBlock/blob/master/xblock/fields.py#L420 because `_field_data_cache` did not contain `grading_policy`.